pkg/credentialprovider/config.go

```diff
@@ -30,6 +30,13 @@ import (
 	"github.com/golang/glog"
 )
 
+// DockerConfigJson represents ~/.docker/config.json file info
+// see https://github.com/docker/docker/pull/12009
+type DockerConfigJson struct {
+	Auths       DockerConfig      `json:"auths"`
+	HttpHeaders map[string]string `json:"HttpHeaders,omitempty"`
```

Comment:
@deads2k Can you explain what you mean by "...take the HttpHeaders support as a follow on item... without plumbing them through.".